packages/store/src/patterns/encodePassable.js

```diff
@@ -237,34 +272,50 @@ const encodeTagged = (tagged, encodePassable) =>
 const decodeTagged = (encoded, decodePassable) => {
   assert(encoded.startsWith(':'));
   const tagpayload = decodeArray(encoded.substring(1), decodePassable);
-  assert(tagpayload.length === 2, X`expected tag,payload pair: ${encoded}`);
+  tagpayload.length === 2 ||
+    assert.fail(X`expected tag,payload pair: ${encoded}`);
   const [tag, payload] = tagpayload;
   passStyleOf(tag) === 'string' ||
     assert.fail(X`not a valid tagged encoding: ${encoded}`);
   return makeTagged(tag, payload);
 };
 
 /**
- * @typedef {object} EncodeOptionsRecord
- * @property {(remotable: object) => string} encodeRemotable
- * @property {(promise: object) => string} encodePromise
- * @property {(error: object) => string} encodeError
+ * @typedef {object} EncodeOptions
+ * @property {(
+ *   remotable: Remotable,
+ *   encodeRecur: (p: Passable) => string,
+ * ) => string} [encodeRemotable]
+ * @property {(
+ *   promise: Promise,
+ *   encodeRecur: (p: Passable) => string,
+ * ) => string} [encodePromise]
+ * @property {(
+ *   error: Error,
+ *   encodeRecur: (p: Passable) => string,
+ * ) => string} [encodeError]
  */
 
 /**
- * @typedef {Partial<EncodeOptionsRecord>} EncodeOptions
- */
-
-/**
- * @param {EncodeOptions=} encodeOptions
+ * @param {EncodeOptions} [encodeOptions]
  * @returns {(passable: Passable) => string}
  */
+// `yarn lint` complains here but not for equivalent code in agoric-sdk.
+// Also, vscode does not complain. Hence we're using at-ts-ignore rather than
+// at-ts-expect-error. Using at-ts-ignore should also generate a complaint
+// that we should be using at-expect-error, where we would normally need
+// to suppress that error as well. However, perhaps that second error currently
+// happens only in agoric-sdk, but not yet in endo. TODO figure out and fix.
+// @ ts-ignore
 export const makeEncodePassable = ({
-  encodeRemotable = rem => assert.fail(X`remotable unexpected: ${rem}`),
-  encodePromise = prom => assert.fail(X`promise unexpected: ${prom}`),
-  encodeError = err => assert.fail(X`error unexpected: ${err}`),
+  encodeRemotable = (rem, _) => assert.fail(X`remotable unexpected: ${rem}`),
+  encodePromise = (prom, _) => assert.fail(X`promise unexpected: ${prom}`),
+  encodeError = (err, _) => assert.fail(X`error unexpected: ${err}`),
 } = {}) => {
   const encodePassable = passable => {
+    if (ErrorHelper.canBeValid(passable, x => x)) {
```

Comment:
I think this can be simplified as of \@endo/marshal 0.7.6: https://github.com/endojs/endo/commit/053ca12110c706439a5f3611592a0b49cb829ac6
```suggestion
    if (ErrorHelper.canBeValid(passable)) {
```

Reply:
Done